contracts/migration/MigrationController.sol

```diff
@@ -50,12 +54,23 @@ contract MigrationController is IMigrationController, StrategyTypes, StrategyCon
         require(strategy.totalSupply() == 0, "Strategy cannot be migrated to");
         require(amount > 0, "No amount");
         require(lpToken.balanceOf(address(genericRouter)) >= amount, "Wrong balance"); // Funds should have been sent to GenericRouter
+        bool sensitiveToSlippage = _sensitiveToSlippage(address(lpToken));
+        int256 estimateBefore = (sensitiveToSlippage) ? oracle().estimateItem(amount, address(lpToken)) : 0;
```

Comment:
re: why `int256`?

A: the estimate from `estimateItem` may be negative, right? If so, a cast to `uint256` may make the comparison "total >= estimateBefore.mul(slippage).div(DIVISOR)" evaluate to an incorrect value.